custom/enikshay/tasks.py

```diff
@@ -483,25 +484,34 @@ def get_aggregated_scores(self, latest_adherence_date, adherence_schedule_date_s
             end_date=update["aggregated_score_date_calculated"]
         )
 
+        doses_per_week = self.get_doses_per_week()
+        update['expected_doses_taken'] = int(((
+            (update['aggregated_score_date_calculated'] - adherence_schedule_date_start)).days / 7.0
+        ) * doses_per_week)
+
+        update['total_expected_doses_taken'] = self.get_total_expected_doses_taken(adherence_schedule_date_start)
+        return update
+
+    @memoized
+    def get_doses_per_week(self):
         # calculate 'expected_doses_taken' score
         dose_data = self.get_doses_data()
         adherence_schedule_id = self.episode.get_case_property('adherence_schedule_id') or DAILY_SCHEDULE_ID
         doses_per_week = dose_data.get(adherence_schedule_id)
-        if doses_per_week:
-            update['expected_doses_taken'] = int(((
-                (update['aggregated_score_date_calculated'] - adherence_schedule_date_start)).days / 7.0
-            ) * doses_per_week)
-        else:
-            update['expected_doses_taken'] = 0
+        if not doses_per_week:
             soft_assert(notify_admins=True)(
                 True,
```

Comment:
i realize this didn't change in this pr, but setting this True means the soft assert never fires?

Reply:
good catch: https://github.com/dimagi/commcare-hq/pull/17932/commits/cdea40b9a95c59b267cee3be8b2b789cac1205d6